File: api/ts/day6.ts
import { NowRequest, NowResponse } from "@vercel/node";
import { StarResult } from "./shared/data";

export default (request: NowRequest, response: NowResponse) => {
  let input = request.body && request.body.input;
  let responseMessage: StarResult;

  const solve = (): StarResult => {
    if (input) {
      let silver = 0;
      let gold = 0;
      const groups = input.split(",,");
      groups.forEach((group) => {
        // silver part
        const groupSet = new Set();
        const persons = group.split(",");

        // gold part
        let goldAnswers = new Set();
        for (let i = 0; i < persons[0].length; i++) {
          goldAnswers.add(persons[0][i]);
        }

        persons.forEach((answers) => {
          // silver part
          for (let i = 0; i < answers.length; i++) {
            groupSet.add(answers[i]);
          }

          // gold part
          let commonAnswers = new Set();
          let goldAnsArray = Array.from(goldAnswers);
          for (let i = 0; i < goldAnsArray.length; i++) {
            if (answers.includes(goldAnsArray[i]))
              commonAnswers.add(goldAnsArray[i]);
          }
          goldAnswers = commonAnswers;
        });
        silver += groupSet.size;
        gold += goldAnswers.size;
      });

      return { silver: silver.toString(), gold: gold.toString() };
    } else return { silver: "No input", gold: "No input" };
  };

  responseMessage = solve();
  response.status(200).json(responseMessage);
};

File: go/day6.go
package main
import (
	"strconv"
	"strings"
)

// Day6 Compute result for Day6
func Day6(data string) StarResult {
	silver := 0
	gold := 0
	groups := strings.Split(data, ";;")
	for _,group := range groups {
		groupSet := NewSet()
		persons := strings.Split(group, ";")

		goldAnswers := NewSet()
		for i := 0; i < len(persons[0]); i++ {
			goldAnswers.Add(string(persons[0][i]))
		}

		for _,answers := range persons {
			// silver part
			for i := 0; i < len(answers); i++ {
				groupSet.Add(string(answers[i]))
			}
			// gold part
			commonAnswers := NewSet()
			goldAnsArray := goldAnswers.ToArray()
			for _,goldAnswer := range goldAnsArray {
				if strings.Contains(answers, goldAnswer) {
					commonAnswers.Add(goldAnswer)
				}
			}
			goldAnswers = commonAnswers
		}
		silver += groupSet.Length()
		gold += goldAnswers.Length()
	}
	return StarResult{Silver: strconv.Itoa(silver), Gold: strconv.Itoa(gold)}
}